packages/lesswrong/server/callbacks/messageCallbacks.ts

```diff
@@ -1,6 +1,40 @@
 import Conversations from '../../lib/collections/conversations/collection'
+import { SENT_MODERATOR_MESSAGE } from '../../lib/collections/moderatorActions/schema';
+import { userIsAdmin } from '../../lib/vulcan-users';
 import { getCollectionHooks } from '../mutationCallbacks';
+import { createMutator } from '../vulcan-lib';
 
 getCollectionHooks("Messages").createAsync.add(function unArchiveConversations({document}) {
   void Conversations.rawUpdateOne({_id:document.conversationId}, {$set: {archivedByIds: []}});
 });
+
+/**
+ * Creates a moderator action when the first message in a mod conversation is sent to the user
+ * This also adds a note to a user's sunshineNotes
+ */
+getCollectionHooks("Messages").createAsync.add(async function updateUserNotesOnModMessage({ document, currentUser, context }) {
+  const { conversationId } = document;
+  const conversation = await context.loaders.Conversations.load(conversationId);
+  if (conversation.moderator) {
+    const [conversationParticipants, conversationMessageCount] = await Promise.all([
+      context.loaders.Users.loadMany(conversation.participantIds),
+      // No need to fetch more than 2, we only care if this is the first message in the conversation
+      context.Messages.find({ conversationId }, { limit: 2 }).count()
+    ]);
+
+    const nonAdminParticipant = conversationParticipants.find(user => !userIsAdmin(user));
```

Comment:
I guess we are assuming that all mod conversations have only one non-admin participant? Seems likely